iree/hal/executable_cache.h

```diff
@@ -98,6 +98,12 @@ typedef struct iree_hal_executable_params_t {
   // executable layout objects.
   iree_host_size_t executable_layout_count;
   iree_hal_executable_layout_t* const* executable_layouts;
+
+  // Executable-level specialization constants table.
+  // The compiler defines the contents of the table, how they are populated, and
+  // their usage in the executable.
+  iree_host_size_t constant_count;
+  const uint32_t* constants;
```

Comment:
Is `executable_params.constants` descriptive enough here? Maybe `spec_constants` or `specialization_constants`? Out of context (without knowing about specialization constants in SPIR-V/Metal) it isn't clear how these fields are used, besides "the compiler wants them". Something in here could either elaborate
* `these allow the executable to be specialized between offline compilation and runtime instantiation`
* `Optional constants provide for specialization of the executable based on runtime-derived parameters.` (copied from compiler code in a future commit)


or link to a reference like 

https://www.khronos.org/registry/vulkan/specs/1.1-khr-extensions/html/chap10.html#pipelines-specialization-constants
https://veldrid.dev/articles/specialization-constants.html

Reply:
good point - I'll add the references to what they are in various backends (always a fan of having a rosetta stone)